Mathlib/CategoryTheory/MorphismProperty/Ind.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2025 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.Limits.Preserves.Over
+import Mathlib.CategoryTheory.MorphismProperty.Comma
+import Mathlib.CategoryTheory.MorphismProperty.Limits
+import Mathlib.CategoryTheory.ObjectProperty.Ind
+
+/-!
+# Ind and pro-properties
+
+Given a morphism property `P`, we define a morphism property `ind P` that is satisfied for
+`f : X ⟶ Y` if `Y` is a filtered colimit of `Yᵢ` and `fᵢ : X ⟶ Yᵢ` satisfy `P`.
+
+We show that `ind P` inherits stability properties from `P`.
+
+## Main definitions
+
+- `CategoryTheory.MorphismProperty.ind`: `f` satisfies `ind P` if `f` is a filtered colimit of
+  morphisms in `P`.
+- `CategoryTheory.MorphismProperty.compact`: `f : X ⟶ Y` is compact if `Y` viewed as an object of
+  `Under X` is compact.
+
+## Main results:
+
+- `CategoryTheory.MorphismProperty.ind_ind`: If `P` implies compact, then `P.ind.ind = P.ind`.
+
+## TODOs:
+
+- Dualise to obtain `pro P`.
+- Show `ind P` is stable under composition if `P` spreads out (Christian).
+-/
+
+universe v u
+
+namespace CategoryTheory.MorphismProperty
+
+open Limits Opposite
+
+variable {C : Type u} [Category.{v} C] (P : MorphismProperty C)
+
+/--
+Let `P` be a property of morphisms. `P.Ind` is satisfied for `f : X ⟶ Y`
+if there exists a family of natural maps `tᵢ : X ⟶ Yᵢ` and `sᵢ : Yᵢ ⟶ Y` indexed by `J`
+such that
+- `J` is filtered
+- `Y = colim Yᵢ` via `{sᵢ}ᵢ`
+- `tᵢ` satisfies `P` for all `i`
+- `f = tᵢ ≫ sᵢ` for all `i`.
+
+See `CategoryTheory.MorphismProperty.ind_iff_ind_under_mk` for an equivalent characterization
+in terms of `Y` as an object of `Under X`.
+-/
+def ind (P : MorphismProperty C) : MorphismProperty C :=
+  fun X Y f ↦ ∃ (J : Type v) (_ : SmallCategory J) (_ : IsFiltered J)
+    (D : J ⥤ C) (t : (Functor.const J).obj X ⟶ D) (s : D ⟶ (Functor.const J).obj Y)
+    (_ : IsColimit (Cocone.mk _ s)), ∀ j, P (t.app j) ∧ t.app j ≫ s.app j = f
+
+variable (C) in
+/-- A morphism `f : X ⟶ Y` is compact if `Y` is compact viewed as an object of `Under X`. -/
+def compact : MorphismProperty C :=
```

Comment:
Similarly as for the object property, I would prefer the name. However, it is a standard definition in the literature. Otherwise, we may clarify things by adding `Under` in the name.